api/verify.go

```diff
@@ -242,3 +245,47 @@ func (a *API) prepErrorRedirectURL(err *HTTPError, r *http.Request) string {
 	q.Set("error_description", err.Message)
 	return rurl + "#" + q.Encode()
 }
+
+func (a *API) emailChangeVerify(ctx context.Context, conn *storage.Connection, params *VerifyParams) (*models.User, error) {
+	instanceID := getInstanceID(ctx)
+	config := a.getConfig(ctx)
+	user, err := models.FindUserByEmailChangeToken(conn, params.Token)
+	if err != nil {
+		if models.IsNotFoundError(err) {
+			return nil, notFoundError(err.Error()).WithInternalError(redirectWithQueryError)
+		}
+		return nil, internalServerError("Database error finding user").WithInternalError(err)
+	}
+
+	nextDay := user.EmailChangeSentAt.Add(24 * time.Hour)
+	if user.EmailChangeSentAt != nil && time.Now().After(nextDay) {
+		return nil, expiredTokenError("Recovery token expired").WithInternalError(redirectWithQueryError)
+	}
+
+	err = a.db.Transaction(func(tx *storage.Connection) error {
+		var terr error
+
+		if params.Token != user.EmailChangeToken {
+			return unauthorizedError("Email Change Token didn't match token on file")
+		}
```

Comment:
We don't have to do this check here because we already do a `models.FindUserByEmailChangeToken` in line 252. If the token is invalid (doesn't match the email change token), then line 252 would return a `models.IsNotFoundError`